src/wazuh_modules/vulnerability_scanner/src/scanOrchestrator/buildSingleAgentListContext.hpp

```diff
@@ -44,8 +46,33 @@ class TBuildSingleAgentListInfoContext final : public AbstractHandler<std::share
      */
     std::shared_ptr<TScanContext> handleRequest(std::shared_ptr<TScanContext> data) override
     {
-        data->m_agents.push_back(
-            {data->agentId().data(), data->agentName().data(), data->agentVersion().data(), data->agentIp().data()});
+        nlohmann::json response;
+
+        try
+        {
+            // Execute query
+            TSocketDBWrapper::instance().query(
+                WazuhDBQueryBuilder::builder()
+                    .globalGetCommand(std::string("agent-info ") + data->agentId().data())
+                    .build(),
+                response);
+        }
+        // LCOV_EXCL_START
+        catch (const std::exception& e)
+        {
+            logDebug2(WM_VULNSCAN_LOGTAG, "Error executing query to fetch global agent data. Reason: %s.", e.what());
+            throw WdbDataException("Error executing query to fetch global agent data", data->agentId().data());
+        }
```

Comment:
We can add coverage to this with the following test

```C++
TEST_F(BuildSingleAgentListContextTest, ExceptionOnDB)
{
    spSocketDBWrapperMock = std::make_shared<MockSocketDBWrapper>();

    EXPECT_CALL(*spSocketDBWrapperMock, query(testing::_, testing::_))
        .Times(1)
        .WillOnce(testing::Throw(SocketDbWrapperException("Error on DB")));

    const auto jsonData = R"(
        {
            "agent_info": {
                "agent_id":"001"
            },
            "action":"upgradeAgentDB"
        })"_json;

    std::variant<const SyscollectorDeltas::Delta*, const SyscollectorSynchronization::SyncMsg*, const nlohmann::json*>
        data = &jsonData;

    auto singleAgentContext = std::make_shared<
        TBuildSingleAgentListInfoContext<TScanContext<TrampolineOsDataCache>, TrampolineSocketDBWrapper>>();

    auto scanContext = std::make_shared<TScanContext<TrampolineOsDataCache>>(data);

    // Context is not used
    EXPECT_THROW(singleAgentContext->handleRequest(scanContext), WdbDataException);

    spSocketDBWrapperMock.reset();
}
```

Reply:
yes, sure